packages/engine/src/avatar/functions/spawnAvatarReceptor.ts

```diff
@@ -124,21 +125,33 @@ export const createAvatarCollider = (entity: Entity) => {
   const colliderEntity = createEntity()
   setComponent(entity, AvatarColliderComponent, { colliderEntity })
 
-  const avatarComponent = getComponent(entity, AvatarComponent)
-  const halfHeight = avatarComponent.avatarHeight * 0.5
-
+  setAvatarColliderTransform(colliderEntity)
   setComponent(colliderEntity, EntityTreeComponent, { parentEntity: entity })
-  setComponent(colliderEntity, TransformComponent, {
-    position: new Vector3(0, halfHeight + 0.25, 0),
-    scale: new Vector3(avatarRadius, halfHeight - avatarRadius - 0.25, avatarRadius)
-  })
   setComponent(colliderEntity, ColliderComponent, {
     shape: Shapes.Capsule,
     collisionLayer: CollisionGroups.Avatars,
     collisionMask: AvatarCollisionMask
   })
 }
 
+const avatarCapsuleOffset = 0.25
+export const setAvatarColliderTransform = (entity: Entity) => {
+  const avatarCollider = getComponent(entity, AvatarColliderComponent)
+  if (avatarCollider === undefined) {
```

Comment:
we should use getOptionalComponent if we are going to check existence, and we can use `!avatarCollider` instead of `=== undefined`

Reply:
implemented